include/libp2p/transport/tcp/tcp_util.hpp

```diff
@@ -56,10 +56,33 @@ namespace libp2p::transport::detail {
 
   inline bool supportsIpTcp(const multi::Multiaddress &ma) {
     using P = multi::Protocol::Code;
-    return (ma.hasProtocol(P::IP4) || ma.hasProtocol(P::IP6))
+    return (ma.hasProtocol(P::IP4) || ma.hasProtocol(P::IP6)
+            || ma.hasProtocol(P::DNS4) || ma.hasProtocol(P::DNS6)
+            || ma.hasProtocol(P::DNS))
         && ma.hasProtocol(P::TCP);
   }
 
+  inline auto getFirstProtocol(const multi::Multiaddress &ma) {
+    return ma.getProtocolsWithValues().front().first.code;
+  }
+
+  // Obtain host and port strings from provided address
+  inline std::pair<std::string, std::string> getHostAndPort(
```

Comment:
As I understand from code, it gets only TCP port. Isn't it somewhat misleading?

Reply:
Don't understand you. It gets both host and port as the name of the method suggests